### PR TITLE
Prow: Add metal3-io org to triggers config

### PR DESCRIPTION
**Pull request by lentzi90**
This should hopefully make it possible to retrigger GitHub workflows on all repos in metal3-io.

### CODE REVIEW

**Comment by lentzi90** [comment]
/override metal3-ubuntu-e2e-integration-test-main

See example here: https://github.com/kubernetes-sigs/prow/blob/dd5d0eeaa28f8ec693371955e01b5d9b55ad8ad9/pkg/plugins/plugin-config-documented.yaml#L598-L624

I now added the whole metal3-io org. Do we want individual repos instead?

File: prow/manifests/overlays/metal3/plugins.yaml [hunk 1]
```diff
@@ -154,5 +155,7 @@ override:
   allow_top_level_owners: true
 
 triggers:
-- # Enable prow to trigger GitHub workflows.
+- repos:
+  - metal3-io
```
Comment (tuminoid) [inline prow/manifests/overlays/metal3/plugins.yaml:159]: Can you put entire org there? In test-infra there is always org/repo pair, and the config is named `repos:` which everywhere else means specfic repository?

/hold

Comment (tuminoid) [reply prow/manifests/overlays/metal3/plugins.yaml:159]: Ah, nevermind. It specifically says "or only org" in the example config.

/unhold
/lgtm

**Comment by tuminoid** [comment]
This is now enabling workflow triggering via `/retest` as seen linked PR above. :tada:

**Pull request merged**